src/Writer/Xml/NilEmptyWriter.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace DemandwareXml\Writer\Xml;
+
+class NilEmptyWriter
+{
+    private $writer;
+
+    public function __construct(XmlWriter $writer)
+    {
+        $this->writer = $writer;
+    }
+
+    public function writeElement($name, $content = null): bool
+    {
+        if (XmlFormatter::isEmptyValue($content)) {
+            return $this->writeElementWithAttributes($name);
+        }
+
+        return $this->writer->writeElement($name, $content);
+    }
+
+    public function writeAttribute($name, $value): bool
+    {
+        return $this->writer->writeAttribute($name, $value);
+    }
+
+    public function writeElementWithAttributes($name, $content = null, array $attributes = []): bool
+    {
+        $this->writer->startElement($name);
+
+        foreach ($attributes as $attrName => $attrContent) {
+            $this->writeAttribute($attrName, $attrContent);
+        }
+
+        if (XmlFormatter::isEmptyValue($content)) {
+            $this->writeAttribute('xsi:nil', 'true');
+            $this->writeAttribute('xmlns:xsi', 'http://www.w3.org/2001/XMLSchema-instance');
```

Comment:
Suspect out of scope but http://www.w3.org/2001/XMLSchema-instance is repeated a fair bit around the place and might benefit from centralisation

Reply:
I think it's only used in one place programmatically - the rest of the uses are tests and I'm more comfortable with them being strings.

Reply:
Oh aye in that case I shall shut my fool mouth